lib/infra/infra-stack.ts

```diff
@@ -0,0 +1,469 @@
+            volume: BlockDeviceVolume.ebs(50, { deleteOnTermination: true }),
+          }],
+          init: CloudFormationInit.fromElements(...InfraStack.getCfnInitElement(this, clusterLogGroup, props, 'ml')),
+          initOptions: {
+            ignoreFailures: false,
+          },
+          signals: Signals.waitForAll(),
+        });
+
+        Tags.of(mlNodeAsg).add('role', 'ml-node');
+      }
+    }
+
+    const alb = new NetworkLoadBalancer(this, 'publicNlb', {
+      loadBalancerName: 'opensearch-nlb',
+      vpc: props.vpc,
+      internetFacing: true,
+    });
+
+    if (!props.securityDisabled && !props.minDistribution) {
+      opensearchListener = alb.addListener('opensearch', {
+        port: 443,
+        protocol: Protocol.TCP,
+      });
+
+      dashboardsListener = alb.addListener('dashboards', {
+        port: 8443,
+        protocol: Protocol.TCP,
+      });
+    } else {
+      opensearchListener = alb.addListener('opensearch', {
+        port: 80,
+        protocol: Protocol.TCP,
+      });
+
+      dashboardsListener = alb.addListener('dashboards', {
+        port: 8443,
+        protocol: Protocol.TCP,
+      });
+    }
+
+    opensearchListener.addTargets('opensearchTarget', {
+      port: 9200,
+      targets: [clientNodeAsg],
+    });
+
+    dashboardsListener.addTargets('dashboardsTarget', {
+      port: 5601,
+      targets: [clientNodeAsg],
+    });
+
+    new CfnOutput(this, 'loadbalancer-url', {
+      value: alb.loadBalancerDnsName,
+      exportName: 'Loadbalancer-URL',
+    });
+  }
+
+  private static getCfnInitElement(scope: Stack, logGroup: LogGroup, props: infraProps, nodeType?: string): InitElement[] {
+    const configFileDir = join(__dirname, '../opensearch-config');
+    let opensearchConfig: string;
+
+    const cfnInitConfig : InitElement[] = [
+      InitPackage.yum('amazon-cloudwatch-agent'),
+      CloudwatchAgent.asInitFile('/opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.json',
+        {
+          agent: {
+            metrics_collection_interval: 60,
+            logfile: '/opt/aws/amazon-cloudwatch-agent/logs/amazon-cloudwatch-agent.log',
+            omit_hostname: true,
+            debug: false,
+          },
+          logs: {
+            logs_collected: {
+              files: {
+                collect_list: [
+                  {
+                    file_path: `/home/ec2-user/opensearch/logs/${scope.stackName}-${scope.account}-${scope.region}.log`,
+                    log_group_name: `${logGroup.logGroupName.toString()}`,
+                    // eslint-disable-next-line no-template-curly-in-string
+                    log_stream_name: '{instance_id}',
+                    auto_removal: true,
+                  },
+                ],
+              },
+            },
+            force_flush_interval: 5,
+          },
+        }),
+      InitCommand.shellCommand('set -ex;/opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a stop'),
```

Comment:
I believe all shell commands have `set -xe` set by default. Can you check and remove from everywhere if not required?

Reply:
Already tested, needs to be present for each statement for it to be reflected in the init log, setting it on top did not log all the commands.